fixtures/attribute-behavior/src/App.js

```diff
@@ -2538,81 +2542,121 @@ function getRenderedAttributeValue(renderer, attribute, type) {
-  let canonicalDefaultValue;
-  try {
-    const read = attribute.read || getProperty(attribute.name);
-
-    testValue = type.testValue;
-    if (attribute.overrideStringValue !== undefined) {
-      switch (type.name) {
-        case 'string':
-          testValue = attribute.overrideStringValue;
-          break;
-        case 'array with string':
-          testValue = [attribute.overrideStringValue];
-          break;
-        default:
-          break;
-      }
+  const read = attribute.read || getProperty(attribute.name);
+  let testValue = type.testValue;
+  if (attribute.overrideStringValue !== undefined) {
+    switch (type.name) {
+      case 'string':
+        testValue = attribute.overrideStringValue;
+        break;
+      case 'array with string':
+        testValue = [attribute.overrideStringValue];
+        break;
+      default:
+        break;
     }
+  }
+  let baseProps = {};
+  if (attribute.type) {
+    baseProps.type = attribute.type;
+  }
+  const props = {
+    ...baseProps,
+    [attribute.name]: testValue,
+  };
 
-    let baseProps = {};
-    if (attribute.type) {
-      baseProps.type = attribute.type;
-    }
+  let defaultValue;
+  let result;
+  let ssrResult;
+  let didWarn;
+  let didError;
+  let ssrDidWarn;
+  let ssrDidError;
+
+  _didWarn = false;
+  try {
     renderer.render(React.createElement(tagName, baseProps), container);
     defaultValue = read(container.firstChild);
-    canonicalDefaultValue = getCanonicalizedValue(defaultValue);
-
-    const props = {
-      ...baseProps,
-      [attribute.name]: testValue,
-    };
     renderer.render(React.createElement(tagName, props), container);
+    result = read(container.firstChild);
+    didWarn = _didWarn;
+    didError = false;
+  } catch (error) {
+    result = null;
+    didWarn = _didWarn;
+    didError = true;
+  }
 
-    const result = read(container.firstChild);
-
-    return {
-      tagName,
-      containerTagName,
-      testValue,
-      defaultValue,
-      result,
-      canonicalResult: getCanonicalizedValue(result),
-      canonicalDefaultValue,
-      didWarn: _didWarn,
-      didError: false,
-    };
+  _didWarn = false;
+  try {
+    const html = serverRenderer.renderToString(
+      React.createElement(tagName, props),
+    );
+    container.innerHTML = html;
+    ssrResult = read(container.firstChild);
+    ssrDidWarn = _didWarn;
+    ssrDidError = false;
   } catch (error) {
-    return {
-      tagName,
-      containerTagName,
-      testValue,
-      defaultValue,
-      result: null,
-      canonicalResult: getCanonicalizedValue(null),
-      canonicalDefaultValue,
-      didWarn: _didWarn,
-      didError: true,
-    };
-  } finally {
-    console.error = originalConsoleError;
+    ssrResult = null;
+    ssrDidWarn = _didWarn;
+    ssrDidError = true;
   }
+
+  console.error = originalConsoleError;
+
+  const canonicalResult = getCanonicalizedValue(result);
+  const canonicalSsrResult = getCanonicalizedValue(ssrResult);
```

Comment:
You need to canonicalize as early as possible because some of these (like SVG) are objects that get mutated so the result could've changed.